README.md

```diff
@@ -137,15 +137,27 @@ from the HardFault handler causes the processor to jump to the start of the Hard
 On hardware, these instructions make the processor sleep until an interrupt signal arrives (WFI) or
 until an event occurs (WFE).
 
-## Unresolved questions
+## How to exit QEMU cleanly?
 
-If you figure these out send a PR with the answer!
+Build with 'semihosting' feature.
+This will pull in the `cortex-m-semihosting` dependency and uses syscall to exit QEMU successfully
+```
+$ cargo build --features semihosting
+```
 
-- You can terminate QEMU from the terminal by entering Ctrl-A + X, but how can you terminate QEMU
-  from within the kernel program?
+Run without `-gdb` and `-S` (stop) option.
+```
+$ qemu-system-arm \
+       -cpu cortex-m3 \
+       -machine lm3s6965evb \
+       -semihosting \
```

Comment:
I have found that `-semihosting-config enable=on,target=native` works better when hooking a debugger (in particular LLDB) to QEMU. Let's recommend that flag instead of `-semihosting`.

Reply:
fix done